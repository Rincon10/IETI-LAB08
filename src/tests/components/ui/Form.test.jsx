import { mount } from 'enzyme';
import { describe, expect, test } from '@jest/globals';
import { UserContext } from 'contexts/UserContext';
import Form from 'components/ui/Form';

describe('Test to the <Form/> component', () => {
    const dispatch = jest.fn();
    var wrapper = mount(<Form />, {
        wrappingComponent: UserContext.Provider,
        wrappingComponentProps: {
            value: { dispatch },
        },
    });

    // before running each test
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = mount(<Form />, {
            wrappingComponent: UserContext.Provider,
            wrappingComponentProps: {
                value: { dispatch },
            },
        });
    });

    test('should render the <Form/> component propertly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should set a default title', () => {
        const expectedTitle = 'Default title';
        const renderedTitle = wrapper.find('h1').text();

        expect(expectedTitle).toBe(renderedTitle);
    });

    test('should render with login title', () => {
        const expectedTitle = 'Login';
        wrapper = mount(<Form title={expectedTitle} />, {
            wrappingComponent: UserContext.Provider,
            wrappingComponentProps: {
                value: { dispatch },
            },
        });

        const renderedTitle = wrapper.find('h1').text();

        expect(expectedTitle).toBe(renderedTitle);
    });
});
