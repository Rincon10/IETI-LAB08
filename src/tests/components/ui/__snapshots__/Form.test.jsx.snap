// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to the <Form/> component should render the <Form/> component propertly 1`] = `
Array [
  Array [
    "",
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1hah0mo-MuiAvatar-root"
    >
      Array [
        "",
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LockOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>,
      ]
    </div>,
  ],
  Array [
    "",
    <h1
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Default title
    </h1>,
  ],
  Array [
    "",
    <form
      className="MuiBox-root css-164r41r"
      noValidate={true}
      onSubmit={[Function]}
    >
      Array [
        "",
        Array [
          "",
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            Array [
              "",
              Array [
                "",
                <label
                  className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={false}
                  htmlFor="email"
                  id="email-label"
                >
                  Email Address
                  Array [
                    "",
                    <span
                      aria-hidden={true}
                      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>,
                  ]
                </label>,
              ],
            ]
            Array [
              "",
              Array [
                "",
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  Array [
                    "",
                    <input
                      aria-invalid={false}
                      autoFocus={true}
                      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={false}
                      id="email"
                      name="email"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="email"
                    />,
                  ]
                  Array [
                    "",
                    Array [
                      "",
                      <fieldset
                        aria-hidden={true}
                        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        Array [
                          "",
                          <legend
                            className="css-173wfxe"
                          >
                            <span>
                              Email Address
                               
                              *
                            </span>
                          </legend>,
                        ]
                      </fieldset>,
                    ],
                  ]
                </div>,
              ],
            ]
          </div>,
        ],
      ]
      Array [
        "",
        Array [
          "",
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            Array [
              "",
              Array [
                "",
                <label
                  className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={false}
                  htmlFor="password"
                  id="password-label"
                >
                  Password
                  Array [
                    "",
                    <span
                      aria-hidden={true}
                      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>,
                  ]
                </label>,
              ],
            ]
            Array [
              "",
              Array [
                "",
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  Array [
                    "",
                    <input
                      aria-invalid={false}
                      autoFocus={true}
                      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={false}
                      id="password"
                      name="password"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="password"
                    />,
                  ]
                  Array [
                    "",
                    Array [
                      "",
                      <fieldset
                        aria-hidden={true}
                        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        Array [
                          "",
                          <legend
                            className="css-173wfxe"
                          >
                            <span>
                              Password
                               
                              *
                            </span>
                          </legend>,
                        ]
                      </fieldset>,
                    ],
                  ]
                </div>,
              ],
            ]
          </div>,
        ],
      ]
      Array [
        "",
        Array [
          "",
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "backgroundColor": "#f00",
              }
            }
            tabIndex={0}
            type="submit"
          >
            Log In
            Array [
              "",
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />,
            ]
          </button>,
        ],
      ]
      Array [
        "",
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          Array [
            "",
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
            >
              Array [
                "",
                Array [
                  "",
                  <a
                    className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-al1l0i-MuiTypography-root-MuiLink-root"
                    href="#"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    Forgot password?
                  </a>,
                ],
              ]
            </div>,
          ]
          Array [
            "",
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
            >
              Array [
                "",
                Array [
                  "",
                  <a
                    className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-al1l0i-MuiTypography-root-MuiLink-root"
                    href="#"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    Don't have an account? Sign Up
                  </a>,
                ],
              ]
            </div>,
          ]
        </div>,
      ]
    </form>,
  ],
]
`;
